src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -249,25 +240,48 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
 
     if( isFieldNameFound == 0 )
     {
-      std::ostringstream fieldNameNotFoundMessage;
-      std::string fieldNamePath =
-        GEOS_FMT( "\n{}: there is no {} named `{}` under the region `{}`.\n",
-                  fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::fieldNameString() ),
-                  FieldSpecificationBase::viewKeyStruct::fieldNameString(),
-                  fs.getFieldName(), fs.getObjectPath() );
-
-      fieldNameNotFoundMessage << fieldNamePath;
-      if( areAllSetsEmpty )
+      std::ostringstream errorMessageBuilder;
+      errorMessageBuilder << GEOS_FMT( "\n{}: there are no {} named `{}` under the region `{}`.\n",
+                                       fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::fieldNameString() ),
+                                       FieldSpecificationBase::viewKeyStruct::fieldNameString(),
+                                       fs.getFieldName(), fs.getObjectPath() );;
+
+      string_array const splitPath =  stringutilities::splitStringWithDelim( fs.getObjectPath(), '/' );
```

Comment:
I'm not sure about these path manipulations. Feels like something that should be handled by `dataRepository`. Is it possible to use
```cpp
template< typename REGIONTYPE = ElementRegionBase, typename ... REGIONTYPES, typename LOOKUP_CONTAINER, typename LAMBDA >
void ElementRegionManager::forElementRegions( LOOKUP_CONTAINER const & targetRegions, LAMBDA && lambda ) const
```